substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,216 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+mod types;
+pub use pallet_scheduler as Schedule;
+pub use types::*;
+
+
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use super::*;
+	use frame_system::WeightInfo;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeCall: Parameter
+			+ Dispatchable<RuntimeOrigin = Self::RuntimeOrigin>
+			+ From<Call<Self>>
+			+ IsType<<Self as frame_system::Config>::RuntimeCall>
+			+ From<frame_system::Call<Self>>;
+
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Type to access the Balances Pallet.
+		type NativeBalance: fungible::Inspect<Self::AccountId>
+			+ fungible::Mutate<Self::AccountId>;
+
+		/// Provider for the block number.
+		type BlockNumberProvider: BlockNumberProvider<BlockNumber = BlockNumberFor<Self>>;
```

Comment:
I don't think it makes sense to force the block number type to be system block number type.
The block number provider should be free to use the relay chain number type or any type.
```suggestion
		type BlockNumberProvider: BlockNumberProvider;
```

you probably want to not import `BlockNumberFor` and instead have a new type alias 
`type ProvidedBlockNumberFor<T> = <<T as Config>::BlockNumberProvider>::BlockNumber;`
or
`type BlockNumberFor<T> = <<T as Config>::BlockNumberProvider>::BlockNumber;`